Provider/src/FirebirdSql.Data.FirebirdClient/Client/Managed/Version10/GdsEventManager.cs

```diff
@@ -60,18 +57,15 @@ public GdsEventManager(int handle, string ipAddress, int portNumber)
 
 		public void QueueEvents(RemoteEvent remoteEvent)
 		{
-			lock (this)
-			{
-				_events[remoteEvent.LocalId] = remoteEvent;
+			_events[remoteEvent.LocalId] = remoteEvent;
 
 #warning Jiri Cincura: I'm pretty sure this is a race condition.
```

Comment:
I have already working piece where the thread is removed completely and non-blocking IO is used instead. Still some work left though. We'll test it heavily with @gonline in coming weeks.